File: backend/app/models/portfolio.rb
class Portfolio < ApplicationRecord
  belongs_to :user
  has_many :features, dependent: :destroy
  has_many :pages, dependent: :destroy
  has_many :images, dependent: :destroy
  has_many :tech_stacks, dependent: :destroy
  has_many :portfolio_tags, dependent: :destroy
  has_many :tags, through: :portfolio_tags
end

File: frontend/src/types/portfolio.ts
import { Feature } from './feature'
import { Image } from './image'
import { Page } from './page'
import { Tag } from './tag'
import { TechStack } from './tech-stack'

export type Portfolio = {
  id: number
  userId: number
  name: string
  thumbnail: string
  description: string
  commitment: string
  challenges: string
  demoUrl: string
  githubUrl: string
  blogUrl: string
  duration: number
  images: Image[]
  techStacks: TechStack[]
  features: Feature[]
  pages: Page[]
  tags: Tag[]
}
